completions/rsync

```diff
@@ -37,32 +37,11 @@ _rsync()
 
     case $cur in
         -*)
-            COMPREPLY=($(compgen -W '--verbose --quiet --no-motd --checksum
-                --archive --recursive --relative --no-implied-dirs
-                --backup --backup-dir= --suffix= --update --inplace --append
-                --append-verify --dirs --old-dirs --links --copy-links
-                --copy-unsafe-links --safe-links --copy-dirlinks
-                --keep-dirlinks --hard-links --perms --executability --chmod=
-                --acls --xattrs --owner --group --devices --copy-devices
-                --specials --times --omit-dir-times --super --fake-super
-                --sparse --dry-run --whole-file --no-whole-file
-                --one-file-system --block-size= --rsh= --rsync-path=
-                --existing --ignore-existing --remove-source-files --delete
-                --delete-before --delete-during --delete-delay --delete-after
-                --delete-excluded --ignore-errors --force --max-delete=
-                --max-size= --min-size= --partial --partial-dir=
-                --delay-updates --prune-empty-dirs --numeric-ids --timeout=
-                --contimeout= --ignore-times --size-only --modify-window=
-                --temp-dir= --fuzzy --compare-dest= --copy-dest= --link-dest=
-                --compress --compress-level= --skip-compress= --cvs-exclude
-                --filter= --exclude= --exclude-from= --include= --include-from=
-                --files-from= --from0 --protect-args --address= --port=
-                --sockopts= --blocking-io --no-blocking-io --stats
-                --8-bit-output --human-readable --progress --itemize-changes
-                --out-format= --log-file= --log-file-format= --password-file=
-                --list-only --bwlimit= --write-batch= --only-write-batch=
-                --read-batch= --protocol= --iconv= --ipv4 --ipv6 --version
-                --help --daemon --config= --no-detach' -- "$cur"))
+            COMPREPLY=($(
+                compgen -W '$(_parse_help "$1")
```

Comment:
Heh, interestingly, the line containing `--help` is formatted differently from everything else here, throwing off `_parse_help` :P. One way to fix it:

```suggestion
                # sed preprocessing for lines like '(-h) --help ...', seen
                # e.g. with rsync 3.1.3 on Ubuntu 20.04
                compgen -W '$(
                    rsync --help 2>&1 | sed -e "s/^([^)]*)//" | _parse_help -
                )
```

Another would be to generalize the fix by doing it in `__parse_options` (likely in a separate PR).

Reply:
Any idea why I can't reproduce this locally when I run `pytest -vv test/t/test_rsync.py::TestRsync`? Tests pass which seems to indicate that `--help, -h (*)` is not a problem and I don't understand why they do if you say we need to pre-process the output.

Reply:
It depends on the version of `rsync`.

```bash
[murase@padparadscha 0 ~]$ rsync --version | head -2
rsync  version 3.1.3  protocol version 31
Copyright (C) 1996-2018 by Andrew Tridgell, Wayne Davison, and others.
[murase@padparadscha 0 ~]$ rsync --help | grep help
(-h) --help                  show this help (-h is --help only if used alone)
Use "rsync --daemon --help" to see the daemon-mode command-line options.
```


```bash
[murase@chatoyancy 0 ~]$ rsync --version | head -2
rsync  version 3.2.3  protocol version 31
Copyright (C) 1996-2020 by Andrew Tridgell, Wayne Davison, and others.
[murase@chatoyancy 0 ~]$ rsync --help | grep help
--help, -h (*)           show this help (* -h is help only on its own)
Use "rsync --daemon --help" to see the daemon-mode command-line options.
```

Reply:
Ah thanks, I'd somehow misread the regex and I thought the newest version had the problem. Looks like https://github.com/WayneD/rsync/commit/cba00be6 fixed it.

I've fixed the problem in my latest push.